features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/retrysendmenu/RetrySendMessageMenu.kt

```diff
@@ -0,0 +1,134 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.timeline.components.retrysendmenu
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.ListItemDefaults
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.element.android.libraries.designsystem.preview.ElementPreviewLight
+import io.element.android.libraries.designsystem.theme.LocalColors
+import io.element.android.libraries.designsystem.theme.components.Text
+import kotlinx.coroutines.launch
+
+@Composable
+internal fun RetrySendMessageMenu(
+    state: RetrySendMenuState,
+    modifier: Modifier = Modifier,
+) {
+    val isVisible = state.selectedEvent != null
+
+    fun onDismiss() {
+        state.eventSink(RetrySendMenuEvents.Dismiss)
+    }
+
+    fun onRetry() {
+        state.eventSink(RetrySendMenuEvents.RetrySend)
+    }
+
+    fun onRemoveFailed() {
+        state.eventSink(RetrySendMenuEvents.RemoveFailed)
+    }
+
+    RetrySendMessageMenuContents(
+        modifier = modifier,
+        isVisible = isVisible,
+        onRetry = ::onRetry,
+        onRemoveFailed = ::onRemoveFailed,
+        onDismiss = ::onDismiss
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun RetrySendMessageMenuContents(
+    isVisible: Boolean,
+    onRetry: () -> Unit,
+    onRemoveFailed: () -> Unit,
+    onDismiss: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val sheetState = rememberModalBottomSheetState()
+    val coroutineScope = rememberCoroutineScope()
+
+    if (isVisible) {
+        ModalBottomSheet(
+            modifier = modifier,
+//            modifier = modifier.navigationBarsPadding() - FIXME after https://issuetracker.google.com/issues/275849044
+//                    .imePadding()
+            sheetState = sheetState,
+            onDismissRequest = {
+                coroutineScope.launch {
+                    sheetState.hide()
+                    onDismiss()
+                }
+            }
+        ) {
+            ListItem(headlineContent = {
+                Text("Your message failed to send", fontWeight = FontWeight.Medium)
```

Comment:
i18n ?

Reply:
Oops. I got so focused on trying to fix the UX issues that I completely forgot to change these placeholders. Thanks!